src/Service/ReceiptHelper.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace App\Service;
+
+use App\Entity\CaseDocumentRelation;
+use App\Entity\CaseEntity;
+use App\Entity\DigitalPost;
+use App\Entity\HearingPost;
+use Doctrine\Bundle\DoctrineBundle\EventSubscriber\EventSubscriberInterface;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Event\LifecycleEventArgs;
+use Doctrine\ORM\Events;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class ReceiptHelper implements EventSubscriberInterface
+{
+    public function __construct(private MailTemplateHelper $mailTemplateHelper, private DocumentUploader $documentUploader, private EntityManagerInterface $entityManager, private DigitalPostHelper $digitalPostHelper, private TranslatorInterface $translator)
+    {
+    }
+
+    public function getSubscribedEvents()
+    {
+        return [
+            Events::postPersist,
+        ];
+    }
+
+    public function postPersist(LifecycleEventArgs $args): void
+    {
+        $entity = $args->getEntity();
+
+        $case = null;
+        $template = null;
+        if ($entity instanceof CaseEntity) {
+            $case = $entity;
+            $digitalPostRecipients[] = (new DigitalPost\Recipient())
+                ->setName($case->getBringer())
+                ->setIdentifierType($case->getBringerIdentification()->getType())
+                ->setIdentifier($case->getBringerIdentification()->getIdentifier())
+                ->setAddress($case->getBringerAddress())
+            ;
+            $template = $case->getBoard()?->getReceiptCase();
+            $documentTitle = $this->translator->trans('Case receipt', [], 'case');
+            // @fixme What should this be? Can we get it from somewhere?
+            $documentType = 'Case receipt';
+        } elseif ($entity instanceof HearingPost) {
+            $hearingPost = $entity;
+            $digitalPostRecipients[] = (new DigitalPost\Recipient())
+                ->setName($hearingPost->getRecipient()->getName())
+                ->setIdentifierType($hearingPost->getRecipient()->getIdentification()->getType())
+                ->setIdentifier($hearingPost->getRecipient()->getIdentification()->getIdentifier())
+                ->setAddress($hearingPost->getRecipient()->getAddress())
+            ;
```

Comment:
`HearingPostResponse` is the entity we're looking for here. This does not have a `recipient` but rather a `sender`. 

`HearingPostRequest` which has a recipient is made by case workers directly in TVIST1 and is then sent via Digital Post to the relevant recipient(s).

I should have named the entities better, this is my bad.